identicon.js

```diff
@@ -118,21 +124,66 @@
         },
 
         // Creates a consistent-length hash from a string
-        createHashFromString: function(str) {
-          var hash = '0', salt = 'identicon', i, chr, len;
+        createHashFromString: function(str){
+            var hash = '0', salt = 'identicon', i, chr, len;
+
+            if (!str) {
+                return hash;
+            }
+
+            str += salt + str; // Better randomization for short inputs.
+
+            for (i = 0, len = str.length; i < len; i++) {
+                chr   = str.charCodeAt(i);
+                hash  = ((hash << 5) - hash) + chr;
+                hash |= 0; // Convert to 32bit integer
+            }
+            return hash.toString();
+        },
+
+        isSvg: function(){
+            return this.format.match(/svg/i)
+        }
+    };
+
+    var Svg = function(size, background){
+        this.size       = size;
+        this.background = this.color.apply(this, background);
+        this.rectangles = [];
+    };
 
-          if (!str) {
-            return hash;
-          }
+    Svg.prototype = {
+        size:       null,
+        background: null,
+        rectangles: null,
+
+        color: function(r, g, b, a){
+            return [r, g, b, a ? a/255 : 1].map(Math.round);
+        },
+
+        getBase64: function(){
+            var i, rect, xml;
+
+            xml = '<svg xmlns="http://www.w3.org/2000/svg"'
+                + ' width="' + this.size + '" height="' + this.size + '"'
+                + ' style="background-color: rgba(' + this.background.join(',') + ');"'
+                + '>';
+
+            for (i = 0; i < this.rectangles.length; i++) {
+                rect = this.rectangles[i];
+                if (rect.color.join(',') == this.background.join(',')) continue;
+                xml += '<rect '
+                    + ' x="' + rect.x + '"'
+                    + ' y="' + rect.y + '"'
+                    + ' width="' + rect.w + '"'
+                    + ' height="' + rect.h + '"'
+                    + ' style="fill: rgba(' + rect.color.join(',') + ');"'
```

Comment:
It might make sense to specify a nonzero stroke width here. Otherwise, I think some browsers draw faint lines between adjacent rectangles. (I've observed such behavior with both Firefox and Chrome on Linux.) See https://github.com/sandstorm-io/sandstorm/blob/v0.187/shell/packages/sandstorm-identicons/identicon.js#L49-L53 .